src/main/java/core/service/impl/ReadService.java

```diff
@@ -1,5 +1,5 @@
 package core.service.impl;
```

Comment:
interface should be in service package and its implementation - in imp packege. You did the opposite